### PR TITLE
helm: renamed podsAnnotations variable into podAnnotations

### PR DESCRIPTION
**Pull request by mvisonneau**
This change follows-up onto https://github.com/cilium/cilium/pull/12189#issuecomment-705226944
It concerns the agent and operator charts

### CODE REVIEW

**Comment by sayboras** [comment]
@mvisonneau thanks, seems like this one is coverred in #13259 as well. If it is already, I think we can wait for a little, as #13259 seems pretty closed from my understanding :)

**Comment by mvisonneau** [comment]
oh cool, sure feel free to do whatever you want with this PR then! :)

**Comment by dominik318** [comment]
@sayboras this PR is required for Cilium DataDog integration which is somewhat urgent for many people. It would be fantastic if we could back-port this to 1.8. This PR is fully backwards compatible.

**Comment by sayboras** [comment]
> this PR is required for Cilium DataDog integration which is somewhat urgent for many people. It would be fantastic if we could back-port this to 1.8. This PR is fully backwards compatible.

Thanks for clarification, I am big fan of of datadog :1st_place_medal:

File: install/kubernetes/cilium/charts/agent/templates/daemonset.yaml
Comment (sayboras) [inline install/kubernetes/cilium/charts/agent/templates/daemonset.yaml:30]: Need other to weigh in, as I am not sure how we capture note for upgrad for this changes. Using `--reuse-values` during upgrade might remove existing pod annotations if this field is not explicitly set in `helm upgrade`

**Review by christarazi — APPROVED** [review]
Thanks! We'll need to wait until the big Helm refactor is merged.

@sayboras I think because the original change never made it into a release, we could probably skip mentioning it. It seems like this PR is just amending the previous change. (Unless I've missed something.)

File: install/kubernetes/cilium/charts/agent/templates/daemonset.yaml
Comment (sayboras) [reply install/kubernetes/cilium/charts/agent/templates/daemonset.yaml:30]: the above is answered by Chris :)

**Review by sayboras — APPROVED** [review]
LGTM 💯

**Review by kaworu — CHANGES_REQUESTED** [review]
Here is my understanding:

https://github.com/cilium/cilium/pull/13259 does include this fix, so the issue will be solved as soon as #13259 is merged into master.

There is a need to have this fix in v1.8. Since this PR won't be merged into master, it could be re-targeted to be merged into the v1.8 branch. Alternatively, this PR could be closed and another PR could be opened against v1.8.

**Comment by joestringer** [comment]
Thanks for clarifying @kAworu , confirmed. @mvisonneau would you mind retargeting these changes for the v1.8 branch?

**Comment by mvisonneau** [comment]
there you go @joestringer! 🙇

**Review by kaworu — APPROVED** [review]
LGTM, thanks!

**Review by rolinh — APPROVED** [review]

**Comment by kAworu** [comment]
test-me-please

**Review by gandro — APPROVED** [review]

**Comment by gandro** [comment]
test-backport-1.8

**Pull request merged**
